src/ripple/app/hook/impl/applyHook.cpp

```diff
@@ -0,0 +1,4976 @@
+
+    assert(kl || hookno);
+
+    // RH UPTO: finish hook_namespace api
+
+    if (!kl)
+    {
+        // special case, asking for self namespace
+        if (aread_ptr == 0)
+        {
+            WRITE_WASM_MEMORY_AND_RETURN(
+                write_ptr, 32,
+                hookCtx.result.hookNamespace.data(), 32,
+                memory, memory_length);
+        }
+
+        if (!hookno)
+            return INVALID_PARAMETERS;
+
+        std::shared_ptr<SLE> hookSLE = applyCtx.view().read(hookCtx.result.);
+        if (!hookSLE || !hookSLE->isFieldPresent(sfHooks))
+            return INTERNAL_ERROR;
+
+        ripple::STArray const& hooks = hookSLE->getFieldArray(sfHooks);
+        if (hook_no >= hooks.size())
+            return DOESNT_EXIST;
+        
+        auto const& hook = hooks[hook_no];
+        if (!hook.isFieldPresent(sfHookHash))
+            return DOESNT_EXIST;
+        }
+
+
+
+    }
+    else
+    if (!kl && acc)
+    {
+    }
+    else
+    if (kl &&
+    {
+    }
+
+
+    
+
+
+    std::shared_ptr<SLE> hookSLE = applyCtx.view().peek(hookCtx.result.hookKeylet);
+    if (!hookSLE || !hookSLE->isFieldPresent(sfHooks))
+        return INTERNAL_ERROR;
+
+    ripple::STArray const& hooks = hookSLE->getFieldArray(sfHooks);
+    if (hook_no >= hooks.size())
+        return DOESNT_EXIST;
+    
+    auto const& hook = hooks[hook_no];
+    if (!hook.isFieldPresent(sfHookHash))
+        return DOESNT_EXIST;
+
+    ripple::uint256 const& hash = hook.getFieldH256(sfHookHash);
+
+    WRITE_WASM_MEMORY_AND_RETURN(
+        write_ptr, write_len,
+        hash.data(), hash.size(),
+        memory, memory_length);
+*/
+}
+
+// Write the account id that the running hook is installed on into write_ptr
+DEFINE_HOOK_FUNCTION(
+    int64_t,
+    hook_account,
+    uint32_t write_ptr, uint32_t ptr_len )
+{
+    HOOK_SETUP(); // populates memory_ctx, memory, memory_length, applyCtx, hookCtx on current stack
+    if (NOT_IN_BOUNDS(write_ptr, 20, memory_length))
+        return OUT_OF_BOUNDS;
+
+    WRITE_WASM_MEMORY_AND_RETURN(
+        write_ptr, 20,
+        hookCtx.result.account.data(), 20,
+        memory, memory_length);
+}
+
+// Deterministic nonces (can be called multiple times)
+// Writes nonce into the write_ptr
+DEFINE_HOOK_FUNCTION(
+    int64_t,
+    etxn_nonce,
+    uint32_t write_ptr, uint32_t write_len )
+{
+    HOOK_SETUP(); // populates memory_ctx, memory, memory_length, applyCtx, hookCtx, view on current stack
+
+    if (write_len < 32)
+        return TOO_SMALL;
+
+    if (NOT_IN_BOUNDS(write_ptr, write_len, memory_length))
+        return OUT_OF_BOUNDS;
+
+    if (hookCtx.emit_nonce_counter > hook_api::max_nonce)
+        return TOO_MANY_NONCES;
+
+
+    // in some cases the same hook might execute multiple times
+    // on one txn, therefore we need to pass this information to the nonce
+    uint32_t flags = 0;
+    flags |= hookCtx.result.isStrong    ? 0b10U: 0;
```

Comment:
I tend to use them because it makes it visually obvious which bit is being set. Obviously there's alternatives like a named bit field or even just passing in bools and an int as additional parameters to the hash function.